### PR TITLE
[MBL-17947][Parent] Fixed student colors and theme colors

### PR DESCRIPTION
**Pull request by tamaskozmer**
Test plan: Test that the app is accessible in both dark mode and light mode. Some notes:
- I fixed the student colors so they have different shade in dark/light mode.
- Fixed canvas brand colors and theme/student color switching.
- Same things apply as for the other apps, when we have a colored background it should have dark text in dark mode, light in light mode and it should be clearly visible.
- I made some refactoring and corrected some other color issues throughout the Parent app.

refs: MBL-17947
affects: Parent
release note: none

## Checklist

- [x] Tested in dark mode
- [x] Tested in light mode
- [x] A11y checked

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by hermannakos — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/parent/src/main/java/com/instructure/parentapp/features/courses/list/CoursesViewModel.kt [hunk 1]
```diff
@@ -55,11 +56,25 @@ class CoursesViewModel @Inject constructor(
...
+
+        viewModelScope.launch {
+            selectedStudentHolder.selectedStudentColorChanged.collect {
+                updateColor()
+            }
+        }
+    }
+
+    private fun updateColor() {
+        _uiState.update {
```
Comment (kristofnemere) [inline apps/parent/src/main/java/com/instructure/parentapp/features/courses/list/CoursesViewModel.kt:68]: Would look better like
```
private fun updateColor() {
     selectedStudent?.let { student ->
         _uiState.update {
             it.copy(studentColor = student.studentColor)
         }
     }
}
```

File: apps/parent/src/main/java/com/instructure/parentapp/features/alerts/settings/AlertSettingsScreen.kt [hunk 2]
```diff
@@ -113,7 +113,10 @@ fun AlertSettingsScreen(
                             }
                         }
                         OverflowMenu(
+                            modifier = Modifier
+                                .background(color = colorResource(id = com.instructure.pandautils.R.color.backgroundLightestElevated)),
```
Comment (kristofnemere) [inline apps/parent/src/main/java/com/instructure/parentapp/features/alerts/settings/AlertSettingsScreen.kt:117]: Could you remove com.instructure.pandautils. ?

File: apps/parent/src/test/java/com/instructure/parentapp/features/splash/SplashViewModelTest.kt
Comment (kristofnemere) [inline apps/parent/src/test/java/com/instructure/parentapp/features/splash/SplashViewModelTest.kt:1]: We could add a test to cover the ApplyTheme event

File: libs/pandautils/src/main/java/com/instructure/pandautils/utils/CanvasContextExtensions.kt [hunk 3]
```diff
@@ -36,7 +36,7 @@ val CanvasContext?.lightColor: Int get() {
 }
 
 @get:ColorInt
-val User?.color: Int get() {
+val User?.studentColor: Int get() {
```
Comment (kristofnemere) [inline libs/pandautils/src/main/java/com/instructure/pandautils/utils/CanvasContextExtensions.kt:39]: Please use this in CourseDetailsViewModel as well

**Review by kristofnemere — CHANGES_REQUESTED** [review]

File: apps/parent/src/test/java/com/instructure/parentapp/features/splash/SplashViewModelTest.kt
Comment (tamaskozmer) [reply apps/parent/src/test/java/com/instructure/parentapp/features/splash/SplashViewModelTest.kt:1]: There is a test for it. See row 105

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/parent/src/main/java/com/instructure/parentapp/features/addstudent/AddStudentViewModel.kt [hunk 4]
```diff
@@ -40,7 +41,7 @@ class AddStudentViewModel @Inject constructor(
     private val _uiState =
         MutableStateFlow(
             AddStudentUiState(
-                color = selectedStudentHolder.selectedStudentState.value.color,
+                color = selectedStudentHolder.selectedStudentState.value.studentColor,
```
Comment (kdeakinstructure) [inline apps/parent/src/main/java/com/instructure/parentapp/features/addstudent/AddStudentViewModel.kt:44]: - [ ] Test

**Review by kristofnemere — APPROVED** [review]
QA+1

**Pull request merged**